src/main/resources/mybatis/collections.xml

```diff
@@ -0,0 +1,117 @@
+    </sql>
+
+    <sql id="buildCollectionLinks">
+        json_build_array(json_build_object(
+             'type', 'application/json',
+             'rel',  'self',
+             'title', 'This document as JSON',
+             'href',  serverUrl || '/api/observations/collections/' || id || '?f=json'
+            ), json_build_object(
+              'type', 'application/geo+json',
+               'rel',  'self',
+               'title', 'Features as GeoJSON',
+               'href',  serverUrl || '/api/observations/collections/' || id || '/items?f=json'
+          )
+        )
+    </sql>
+
+    <sql id="buildCollectionsLinks">
+        json_build_array(json_build_object(
+                'type', 'application/json',
+                'rel',  'self',
+                'title', 'This document as JSON',
+                'href',  #{serverUrl} || '/api/observations/collections?f=json'
+              )
+         )
+    </sql>
+
+    <sql id="collectionCols">
+        'id', collectionId,
+        'itemType', 'feature',
+        'title', title,
+        'description', description,
+        'extent', json_build_object('spatial',
+            json_build_object('bbox', json_build_array(
+              json_build_array( xmin, ymin, xmax, ymax)
+            )
+          ),
+          'crs', 'http://www.opengis.net/def/crs/OGC/1.3/CRS83'
+        ),
+        'links', json_build_array(json_build_object(
+             'type', 'application/json',
+             'rel',  'self',
+             'title', 'This document as JSON',
+             'href',  serverUrl || '/api/observations/collections/' || id || '?f=json'
+            ), json_build_object(
+              'type', 'application/geo+json',
+               'rel',  'self',
+               'title', 'Features as GeoJSON',
+               'href',  serverUrl || '/api/observations/collections/' || id || '/items?f=json'
+          )
+        )
+    </sql>
+
+    <sql id="colsToJsonFieldName">
+         id "collectionId", 'feature' "itemType", title "title", description "description",
+         json_build_array('') keywords,
+         <include refid="buildExtent"/> extent,
+         <include refid="buildCollectionLinks"/> links
+    </sql>
+
+    <select id="getCollectionsJson" parameterType="Map" resultType="String">
+      select json_build_object('collections', json_agg(collections),
+                              'links', <include refid="buildCollectionsLinks"/>)
+        from(select <include refid="colsToJsonFieldName"/> 
+          from (select *
+                        from (<include refid="selectCollections"/>) as a) as jsonFields)
+           as collections
+    </select>
+
+    <select id="getCollectionJson" parameterType="Map" resultType="String">
+      select row_to_json(collection)
+        from(select <include refid="colsToJsonFieldName"/> 
+          from (select *
+                        from (<include refid="selectCollections"/>) as a
+                        where id = #{collectionId}) as jsonFields)
```

Comment:
Having the where clause here could lead to performance problems - entire result set might be materialized before filtering.